include/mbgl/util/string.hpp

```diff
@@ -10,7 +10,16 @@ namespace util {
 
 template <class T>
 inline std::string toString(T t) {
-    return std::to_string(t);
+    if (t - (int)t > 0.0 || t + (int)t < 0.0) {
+        std::string str = std::to_string(t);
+        str.erase(str.find_last_not_of('0') + 1, std::string::npos);
+        if (str.back() == '.') {
```

Comment:
According to the [`std::to_string()`](http://www.cplusplus.com/reference/string/to_string/) documentation, the output matches that of `printf()`. `printf()` is affected by the current locale, which isn’t necessarily the default C locale. If you were to call `std::locale::global(std::locale("fr"))`, the decimal separator would be `,`, not `.`.

I actually have no idea if other parts of mbgl assume the C locale as this code does, but I suspect that would be problematic on some of the platforms we support.